tests/components/advantage_air/test_climate.py

```diff
@@ -265,3 +269,21 @@ async def test_climate_async_failed_update(
         )
 
     mock_update.assert_called_once()
+
+
+async def test_coordinator_refresh(
+    hass: HomeAssistant,
+    mock_get: AsyncMock,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test climate after coordinator refresh."""
+
+    await add_mock_config(hass)
+    freezer.tick(ADVANTAGE_AIR_SYNC_INTERVAL)
+
+    with patch(
+        "homeassistant.components.advantage_air.climate.AdvantageAirAC._async_configure_preset"
+    ) as set_preset:
```

Comment:
I am still missing what we are testing here? We should not patch internal functions

Reply:
The code path I am trying to test is that `_async_configure_preset` gets called when the coordinator refreshes. 
https://github.com/home-assistant/core/blob/dev/homeassistant/components/advantage_air/climate.py#L127

Its not an internal function, just a private function I created in the climate class.

Reply:
We should rather test that async_configure_preset is doing what it is doing. I don't really care that `_async_configure_preset` is called, I care about that waiting x minutes, can have this as result. This way the tests are not tightly coupled to the integration code and leaves more room to changing implementations without touching tests

Reply:
`_async_configure_preset` is used at init and on change, so its functionality is already tested elsewhere.

The intent of the added test is to simply ensure `_async_configure_preset` is also called on coordinator refresh (to cover the value being changed externally).

I could mock the refresh with a different (or modified) fixture, and do a state assertion, if required.